syntax="proto3";
import "google/protobuf/empty.proto";

message MODULE{
    string data =1;
}
service GETTER_INFORMATION{
    rpc UNAME(google.protobuf.Empty) returns (MODULE){}
    rpc MODINFO(MODULE) returns (stream MODULE){}
    rpc CURRENT_CONFIG(google.protobuf.Empty) returns (MODULE){}
    rpc RUNNING_MODULES(google.protobuf.Empty) returns (stream MODULE){}
    rpc ALL_MODULES(google.protobuf.Empty) returns (MODULE){}
    rpc ALL_USER_SPACE_OBJECTS(google.protobuf.Empty) returns (stream MODULE){}
}